src/components/calendars/WeekdayFilterCheckbox.tsx

```diff
@@ -0,0 +1,43 @@
+import { FormControl, InputLabel, Select, MenuItem, Checkbox, ListItemText, FormHelperText } from '@mui/material';
+import { daysOfWeek } from '../../helpers/timeConversions';
+
+interface WeekdayFilterProps {
+  label?: string;
+  value: string[];
+  onChange: any;
+}
+
+const WeekdayFilterCheckbox = (props: WeekdayFilterProps) => {
+
+  const formatDay = (day:string) => {
+    return day.charAt(0).toUpperCase() + day.slice(1).toLowerCase();
+  };
+
+  return (
+    <FormControl variant="standard" fullWidth>
+      <InputLabel>{props.label || "Select at least one weekday"}</InputLabel>
+      <Select
+        multiple
+        value={props.value}
+        onChange={props.onChange}
+        renderValue={(selected) => (
```

Comment:
Let's also extract this as a separate function. The general rule should be the following: if render contains function which is longer than one line, we move it as a standalone function and reference it here.